src/consumer.rs

```diff
@@ -54,11 +55,34 @@ impl ConsumerInternal {
     }
 }
 
+#[derive(Clone)]
+pub struct FilterConfiguration {
+    filter_values: Vec<String>,
+    pub predicate: Arc<dyn Fn(&Message) -> bool + Send + Sync>,
+    match_unfiltered: bool,
+}
+
+impl FilterConfiguration {
+    pub fn new(
+        filter_values: Vec<String>,
+        predicate: impl Fn(&Message) -> bool + 'static + Send + Sync,
```

Comment:
Even if normally it is used to avoid "false positive" I think that predicate is Optional and the user can skip it if not necessary @Gsantomaggio to confirm